forte/base_classes/rdms.h

```diff
@@ -352,8 +358,8 @@ void make_cumulant_L3bbb_in_place(const ambit::Tensor& g1b, const ambit::Tensor&
  * @param Enuc the nucleaer repulsion energy
  * @return the reference energy
  */
-double compute_Eref_from_rdms(RDMs& ref, std::shared_ptr<ForteIntegrals> ints,
-                              std::shared_ptr<MOSpaceInfo> mo_space_info);
+double compute_Eref_from_rdms(RDMs& ref, const std::shared_ptr<ForteIntegrals>& ints,
```

Comment:
I noticed that you use `const std::shared_ptr<T>` here. This just means that the pointer is constant. If you want to enforce that the integral object is constant (e.g. to signal that the orbitals should not change) you need to pass `std::shared_ptr<const T>`. I assume that this would require a type conversion from `const std::shared_ptr<T>` to `std::shared_ptr<const T>`, but I am not sure about how this works.

Reply:
Ok, I found it: "shared_ptr<T> is convertible to shared_ptr<const T> but not the reverse."

Reply:
This is just what CLion suggests for tidiness. I can change them for all at the end.

Reply:
I see, does CLion give you an explanation for the suggestion? For what I see, it is suggesting to protect the shared pointer (which is OK). Probably the best thing to do would be `const std::shared_ptr<const ForteIntegrals>` but this starts to be verbose. :)

Reply:
Actually is from clang-tidy. This is the message I get: `Clang-Tidy: The parameter 'ints' is copied for each invocation but only used as a const reference; consider making it a const reference`